verification.mk

```
@@ -0,0 +1,136 @@
+
+# ========================================================================
+#                          SMART CONTRACT VERIFICATION
+# ========================================================================
+# Chain IDs for supported networks
+SEPOLIA_CHAIN_ID := 11155111
+ARBITRUM_SEPOLIA_CHAIN_ID := 421614
+
+# ========================================================================
+#                          VERIFICATION FUNCTIONS
+# ========================================================================
+
+# Verify ERC1967 proxy contracts
+# Parameters: CONTRACT_NAME, NETWORK, CONFIG_KEY, CHAIN_ID, DISPLAY_NAME
+define verify-proxy
+	@echo "Verifying $(1) Proxy on $(5)..."
+	forge verify-contract \
+		--chain-id $(4) \
+		--watch \
+		--etherscan-api-key $(ETHERSCAN_API_KEY) \
+		$$(./scripts/get_config_address.sh $(2) $(3)) \
+		lib/openzeppelin-contracts/contracts/proxy/ERC1967/ERC1967Proxy.sol:ERC1967Proxy
+	@echo "Proxy verification completed for $(1) on $(5)"
+endef
+
+# Verify implementation contracts with optional constructor arguments
+# Parameters: CONTRACT_NAME, NETWORK, CONFIG_KEY, CHAIN_ID, DISPLAY_NAME, CONTRACT_PATH, RPC_URL, CONSTRUCTOR_ARGS
+define verify-impl
+	@echo "Verifying $(1) Implementation on $(5)..."
+	@proxy_address=$$(./scripts/get_config_address.sh $(2) $(3)); \
+	impl_address=$$(./scripts/get_implementation_address.sh $$proxy_address $(7) | tail -n 1); \
+	echo "Proxy address: $$proxy_address"; \
+	echo "Implementation address: $$impl_address"; \
+	if [ -n "$(8)" ]; then \
+		forge verify-contract \
+			--chain-id $(4) \
+			--watch \
+			--constructor-args $(8) \
```

```suggestion
			$(8) \
```
with 
```
CONSTRUCTOR_ARGS="--constructor-args $$constructor_args"
```
and no need for the if/else.

---

[feat: update constructor argument handling in verification scripts](https://github.com/iExecBlockchainComputing/rlc-multichain/pull/80/commits/9cc00a445171e1a3a41ae79f8bf8e157a8556449)
[feat: streamline implementation verification process in verification.mk](https://github.com/iExecBlockchainComputing/rlc-multichain/pull/80/commits/2829f2df5102dc20380d2e85bf18f6627f79aac2)